### ISSUE TITLE
(1d) sequence alteration needs to be taken into account when exporting GFF3 UcA feature, UcA track, and reference sequence in a consistent manner

### ISSUE THREAD

**Issue by deepakunni3** [GMOD/Apollo#420]
#404

**Comment by nathandunn** [comment]
I talked to @monicacecilia  the way it works with the right-click (integrating the sequence alterations into the sequence) is the correct way to do it as we had thought.    We just need to make it work like that with the other exports.   There should already be code in there for "getResiduesWithAlterations" or something like that.

**Comment by selewis** [comment]
Yep, certainly needs this for the affected transcripts (and when we get to
it, for translational frameshifts at the protein level)

Wondering if it would be possible (and advisable) to include a
note/comment/meta about the existence of the alteration(s). FASTA header??
GFF3 column 9???

-S

**Comment by nathandunn** [comment]
This is what it does right now (in WA1):

// normal GFF3 entries above
###
GroupUn4157	.	substitution	216	216	.	+	.	Name=5F68ACD2E557C6E4DCFBD8A79884FCE6;ID=5F68ACD2E557C6E4DCFBD8A79884FCE6
###
GroupUn4157	.	insertion	1	1	.	+	.	Name=819DD4148DE617F4F8585DD4B45CB799;ID=819DD4148DE617F4F8585DD4B45CB799
###
GroupUn4157	.	insertion	1250	1250	.	+	.	Name=881304A2B0E0EA7428BE7A5C323C5C5A;ID=881304A2B0E0EA7428BE7A5C323C5C5A
###
GroupUn4157	.	insertion	675	675	.	+	.	Name=FC750960CACFC7DE990CE5E5EEA8B797;ID=FC750960CACFC7DE990CE5E5EEA8B797
###
// normal FASTA entry goes here 
>5F68ACD2E557C6E4DCFBD8A79884FCE6
A
>819DD4148DE617F4F8585DD4B45CB799
AAAAAAAAA
>881304A2B0E0EA7428BE7A5C323C5C5A
AAAAAAAAAAAAAATTTTTTTTTT
>FC750960CACFC7DE990CE5E5EEA8B797
AAAAAAAAAAAAATTTTTTTT

Also noticed that the get_gff3 for feature does not include alterations like this.  We should be consistent with this. 

Nathan

**Comment by deepakunni3** [comment]
GFF3 export from U-cA and Export Panel provides option for 'GFF3 only' or 'GFF3 with FASTA'.
The latter option includes reference sequence FASTA as part of GFF3 download.

Sequence Alterations are also exported as part of GFF3.

**Comment by nathandunn** [comment]
@deepakunni3   I think that this is ready.

**Comment by nathandunn** [comment]
@deepakunni3   Let me know if its not.

**Comment by deepakunni3** [comment]
@nathandunn This works as expected. Ready for testing.

**Comment by monicacecilia** [comment]
... oh, and this. 
@deepakunni3 @nathandunn  No, sequence alterations are not being reported as part of the GFF3.

Unless I am missing something, I can't find them in there.

**Comment by monicacecilia** [comment]
please let me rephrase this: sequence alterations ARE indeed being taken into account when reporting the GFF3 and FASTA - that is, the exported sequences are modified accordingly. 
However, we have not found indication of sequence alterations being included in the definition line, nor on description columns, etc. 
Am I dreaming, that these things should be reported somewhere?

**Comment by monicacecilia** [comment]
Update: reporting the alterations is failing only when retrieving the GFF3 with "Get GFF3" from the track. 
It is visible when using the "Save data" option from the track label menu, and when saving the GFF3 file from the "Ref Sequence" tab on the annotator panel on the right.

**Comment by monicacecilia** [comment]
Update: sequence alterations are being taken into account and reported in all GFF3 files. :dancer:
